src/cli.ts

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env node
+
+import { addDays, format, isBefore } from "date-fns";
+import prompts, { Answers } from "prompts";
+import { getConfig, overrideConfig } from "./config";
+import { run } from "./shared/leaderboard";
+import { logError } from "./shared/shared-logger";
+
+// `prompts` package uses generics which accepts string literal values and then uses those values as types
+// This means I have to ensure I use a string literal type that matches a string literal value
+const PROMPT_NAME = "value";
+type PromptType = "value";
```

Comment:
This is ugly.  Is there a way around this?

Reply:
Oh nice we can do that

![image](https://user-images.githubusercontent.com/47808687/121361119-5557ad80-c978-11eb-8aed-263ba98f6c39.png)

Reply:
Actually no that won't work, because it makes it impossible to access data from **all** previous prompts.  This then makes it impossible to access things like `startDate` in the `endDate` validator, among other things